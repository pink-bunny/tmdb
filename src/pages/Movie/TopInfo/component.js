import React from 'react';
import {
  Row,
  Col,
  Typography,
  Tag
} from 'antd';
import PropTypes from 'prop-types';

import ToggleToWatchlist from './ToggleToWatchlist';
import ToggleToFavorite from './ToggleToFavorite';
import AddToListPopover from './AddToListPopover';

const TopInfo = ({ details }) => {
  const {
    id,
    title,
    overview,
    original_language: originalLanguage,
    runtime,
    budget,
    revenue,
    genres,
    watchlist,
    favorite
  } = details;

  return (
    <>
      <Row>
        <Col
          span={20}
          offset={2}
        >
          <Typography.Title>
            <span>
              {title}
            </span>
            {' '}
            <AddToListPopover movieId={id} />
            {' '}
            <ToggleToFavorite id={id} favorite={favorite} />
            {' '}
            <ToggleToWatchlist id={id} watchlist={watchlist} />
          </Typography.Title>
          {overview && (
            <>
              <Typography.Title level={3}>Overview</Typography.Title>
              <Typography.Paragraph>
                {overview}
              </Typography.Paragraph>
            </>
          )}
        </Col>
      </Row>
      <Row>
        <Col
          span={20}
          offset={2}
        >
          <Typography.Paragraph>
            <b>Original Language: </b>
            <span>
              {originalLanguage}
            </span>
          </Typography.Paragraph>
        </Col>
        <Col
          span={20}
          offset={2}
        >
          {runtime && (
            <>
              <Typography.Paragraph>
                <b>Runtime: </b>
                <span>
                  {runtime}
                </span>
              </Typography.Paragraph>
            </>
          )}
        </Col>
        <Col
          span={20}
          offset={2}
        >
          <Typography.Paragraph>
            <b>Budget: </b>
            <span>
              $
              {budget}
            </span>
          </Typography.Paragraph>
        </Col>
        <Col
          span={20}
          offset={2}
        >
          <Typography.Paragraph>
            <b>Revenue: </b>
            <span>
              $
              {revenue}
            </span>
          </Typography.Paragraph>
        </Col>
        <Col
          span={20}
          offset={2}
        >
          <Typography.Paragraph>
            <b>Genres: </b>
            {genres.map((item) => (
              <Tag key={item.id}>
                {item.name}
              </Tag>
            ))}
          </Typography.Paragraph>
        </Col>
      </Row>
    </>
  );
};

TopInfo.propTypes = {
  details: PropTypes.shape({
    id: PropTypes.number.isRequired,
    overview: PropTypes.string,
    title: PropTypes.string.isRequired,
    original_language: PropTypes.string.isRequired, /* eslint-disable-line camelcase */
    runtime: PropTypes.number,
    budget: PropTypes.number.isRequired,
    revenue: PropTypes.number.isRequired,
    genres: PropTypes.arrayOf(PropTypes.object).isRequired,
    watchlist: PropTypes.bool.isRequired,
    favorite: PropTypes.bool.isRequired
  })
};

TopInfo.defaultProps = {
  details: {
    overview: null,
    runtime: null
  }
};

export default TopInfo;
