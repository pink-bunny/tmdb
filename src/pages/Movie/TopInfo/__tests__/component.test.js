import React from 'react';
import { shallow } from 'enzyme';

import TopInfo from '../component';

describe('TopInfo component', () => {
  const requiredProps = {
    details: {
      id: 1,
      title: 'Mock title',
      original_language: 'Mock language', /* eslint-disable-line camelcase */
      budget: 1000000,
      revenue: 50000,
      genres: [{ id: 1, name: 'Mock genre' }],
      watchlist: false
    }
  };
  /* eslint-disable react/jsx-props-no-spreading */
  const wrapper = shallow(<TopInfo {...requiredProps} />);
  /* eslint-enable react/jsx-props-no-spreading */

  it('matches snapshot with required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has optional props', () => {
    wrapper.setProps({
      details: {
        ...requiredProps.details,
        overview: 'Mock overview',
        runtime: 124
      }
    });

    expect(wrapper).toMatchSnapshot();
  });
});
