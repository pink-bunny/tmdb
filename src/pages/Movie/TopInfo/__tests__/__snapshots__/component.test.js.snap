// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopInfo component has optional props 1`] = `
<Fragment>
  <Row
    gutter={0}
  >
    <Col
      offset={2}
      span={20}
    >
      <Title>
        <span>
          Mock title
        </span>
         
        <Icon
          type="heart"
        />
         
        <Connect(ToggleToWatchlistContainer)
          id={1}
          watchlist={false}
        />
      </Title>
      <Title
        level={3}
      >
        Overview
      </Title>
      <Paragraph>
        Mock overview
      </Paragraph>
    </Col>
  </Row>
  <Row
    gutter={0}
  >
    <Col
      offset={2}
      span={20}
    >
      <Paragraph>
        <b>
          Original Language: 
        </b>
        <span>
          Mock language
        </span>
      </Paragraph>
    </Col>
    <Col
      offset={2}
      span={20}
    >
      <Paragraph>
        <b>
          Runtime: 
        </b>
        <span>
          124
        </span>
      </Paragraph>
    </Col>
    <Col
      offset={2}
      span={20}
    >
      <Paragraph>
        <b>
          Budget: 
        </b>
        <span>
          $
          1000000
        </span>
      </Paragraph>
    </Col>
    <Col
      offset={2}
      span={20}
    >
      <Paragraph>
        <b>
          Revenue: 
        </b>
        <span>
          $
          50000
        </span>
      </Paragraph>
    </Col>
    <Col
      offset={2}
      span={20}
    >
      <Paragraph>
        <b>
          Genres: 
        </b>
        <Tag
          closable={false}
          key="1"
        >
          Mock genre
        </Tag>
      </Paragraph>
    </Col>
  </Row>
</Fragment>
`;

exports[`TopInfo component matches snapshot with required props 1`] = `
<Fragment>
  <Row
    gutter={0}
  >
    <Col
      offset={2}
      span={20}
    >
      <Title>
        <span>
          Mock title
        </span>
         
        <Icon
          type="heart"
        />
         
        <Connect(ToggleToWatchlistContainer)
          id={1}
          watchlist={false}
        />
      </Title>
    </Col>
  </Row>
  <Row
    gutter={0}
  >
    <Col
      offset={2}
      span={20}
    >
      <Paragraph>
        <b>
          Original Language: 
        </b>
        <span>
          Mock language
        </span>
      </Paragraph>
    </Col>
    <Col
      offset={2}
      span={20}
    />
    <Col
      offset={2}
      span={20}
    >
      <Paragraph>
        <b>
          Budget: 
        </b>
        <span>
          $
          1000000
        </span>
      </Paragraph>
    </Col>
    <Col
      offset={2}
      span={20}
    >
      <Paragraph>
        <b>
          Revenue: 
        </b>
        <span>
          $
          50000
        </span>
      </Paragraph>
    </Col>
    <Col
      offset={2}
      span={20}
    >
      <Paragraph>
        <b>
          Genres: 
        </b>
        <Tag
          closable={false}
          key="1"
        >
          Mock genre
        </Tag>
      </Paragraph>
    </Col>
  </Row>
</Fragment>
`;
